banzai/tic-tac-toe.rb

```diff
@@ -0,0 +1,108 @@
+# this gem injects a color method to the string when it is displayed by console
+require 'colorize' 
+
+@level = 2
+
+def draw_field(val, index)
+  (val.is_a? Integer) ? (index + 1 < 10 ? " #{val}" : val) : " #{val}"
+end
+
+def draw_cell(cell, index)
+  return draw_field(index + 1, index).to_s.white if cell == 0
+  return draw_field("O", index).yellow if cell == -1
+  return draw_field("X", index).blue if cell == 1
+end
+
+def draw_tic(arr)
+  size_line = 5 * @level + 1
+  line, col = "-".green, "|".green
+  border = line * size_line
+  total = border
+  row = col
+  arr.each_with_index do |cell, index|
+    content = draw_cell(cell, index)
+    row = row + " #{content} #{col}"
+    if ((index + 1) % @level == 0)
+      total = total + "\n" + row + "\n" + border
+      row = col
+    end
+  end
+  puts total
+end
+
+def finish_game?(arr)
+  arr.include?(0)
+end
+
+def get_index(arr, val)
+  arr.map.with_index { |num, index| index if num == val}.compact
+end
+
+def who_wins?(g_state)
+  board = [*0..@level**2 -1].each_slice(@level).to_a
```

Comment:
What does the * do at the array start?
Qué hace el * al incio del array?

Reply:
Esta syntasis hace lo mismo que ` (1..9).to_a`

Reply:
Como les mencioné, usar espacios correctos siempre es una buena idea porque les permite leer más fácilmente las expresiones: `[*0..(@level ** 2 - 1)]`